ballerina/response_formatter.bal

```diff
@@ -0,0 +1,199 @@
+                    return ();
+                }
+                result.push(elementResult);
+            } else if element == () {
+                if elementType.kind == NON_NULL {
+                    return ();
+                }
+                result.push(element);
+            } else {
+                result.push(element);
+            }
+        }
+        return result;
+    }
+
+    isolated function coerceObjectField(Data data, parser:FieldNode fieldNode, __Type parentType)
+    returns anydata|anydata[] {
+        __Type objectType = unwrapNonNullype(parentType);
+        anydata|anydata[] fieldValue = data.get(fieldNode.getAlias());
+        if fieldValue == () {
+            return fieldValue;
+        } else if fieldValue is anydata[] {
+            __Type elementType = unwrapNonNullype(parentType);
+            if elementType.kind == LIST {
+                elementType = unwrapNonNullype(<__Type>elementType?.ofType);
+            }
+            __Type fieldType = self.getFieldType(fieldNode.getName(), elementType);
+            return self.coerceArray(fieldValue, fieldNode, fieldType);
+        } else if fieldValue is Data {
+            __Type fieldType = self.getFieldType(fieldNode.getName(), parentType);
+            return self.coerceObject(fieldValue, fieldNode, fieldType);
+        } else {
+            return fieldValue;
+        }
+    }
+
+    isolated function getFieldType(string fieldName, __Type parentType) returns __Type {
+        __Type objectType = getOfType(parentType);
+        __Field selectionField = self.getField(objectType, fieldName);
+        return selectionField.'type;
+    }
+
+    isolated function getField(__Type parentType, string fieldName) returns __Field {
+        if fieldName == SCHEMA_FIELD {
+            __Type fieldType = <__Type>getTypeFromTypeArray(self.schema.types, SCHEMA_TYPE_NAME);
+            return createField(SCHEMA_FIELD, fieldType);
+        } else if fieldName == TYPE_FIELD {
+            __Type fieldType = <__Type>getTypeFromTypeArray(self.schema.types, TYPE_TYPE_NAME);
+            __Type argumentType = <__Type>getTypeFromTypeArray(self.schema.types, STRING);
+            __Type wrapperType = { kind: NON_NULL, ofType: argumentType };
+            __InputValue[] args = [{ name: NAME_ARGUMENT, 'type: wrapperType }];
+            return createField(TYPE_FIELD, fieldType, args);
+        } else if fieldName == TYPE_NAME_FIELD {
+            __Type ofType = <__Type>getTypeFromTypeArray(self.schema.types, STRING);
+            __Type wrappingType = { kind: NON_NULL, ofType: ofType };
+            return createField(TYPE_NAME_FIELD, wrappingType);
+        } else {
+            __Field[] fields = <__Field[]>parentType?.fields;
+            return <__Field>getFieldFromFieldArray(fields, fieldName);
+        }
+    }
+}
+
+isolated function sortErrorDetail(ErrorDetail errorDetail) returns int {
+    Location[]? locations = errorDetail?.locations;
+    if (locations == ()) {
```

Comment:
we can remove the parentheses here. can't we?

Reply:
Fixed with https://github.com/ballerina-platform/module-ballerina-graphql/pull/446/commits/231a8fcb1db1504fed187d2dc1a3957434e6d03e